api4/group.go

```diff
@@ -129,6 +134,13 @@ func patchGroup(c *Context, w http.ResponseWriter, r *http.Request) {
 	auditRec.AddMeta("old_group_display", group.DisplayName)
 	auditRec.AddMeta("old_group_desc", group.Description)
 
+	if groupPatch.AllowReference != nil && *groupPatch.AllowReference {
+		tmp := model.NewId()
```

Comment:
I don't understand the reason for the `NewId()` fallback for the `(*Group).Name` field.

Reply:
The thinking is that currently we don't have a way to set the Name field in the UI, so that if there ever is a spurious call to patch the group witt a path containing a non-nil field, we reset it to the uuid we have as default.
However, in preparation for the moment in which we can set the @slug in the client, we can have instead:
`if groupPatch.AllowReference != nil && *groupPatch.AllowReference {
		if groupPatch.Name == nil {
			tmp := strings.ReplaceAll(group.DisplayName, " ", "-")
			groupPatch.Name = &tmp
		}
	}`